phys2bids/bids_units.py

```diff
@@ -0,0 +1,94 @@
+                # volt: voltage (electrical potential), emf
+                'v': 'V', 'volt': 'V', 'volts': 'V',
+                # degree Celsius: temperature relative to 273.15 K
+                '°c': '°C', 'celsius': '°C', '°celsius': '°C',
+                # second: time
+                'second': 's', 'seconds': 's', 'sec': 's',
+                'secs': 's',
+                '1/hz': 's', '1/hertz': 's', 's': 's'
+}
+
+# Init dictionary of aliases for multipliers. Entries are still lowercase
+prefix_aliases = {
+                    # Multiples - skip "mega" and only up to "tera"
+                    'da': 'da', 'deca': 'da', 'h': 'h', 'hecto': 'h',
+                    'k': 'k', 'kilo': 'k', 'g': 'G', 'giga': 'G', 't': 'T',
+                    'tera': 'T',
+                    # Submultipliers
+                    'd': 'd', 'deci': 'd', 'c': 'c', 'centi': 'c',
+                    'milli': 'm', 'm': 'm', 'µ': 'µ', 'micro': 'µ',
+                    'n': 'n', 'nano': 'n', 'p': 'p', 'pico': 'p',
+                    'f': 'f', 'femto': 'f', 'a': 'a', 'atto': 'a',
+                    'z': 'z', 'zepto': 'z', 'y': 'y', 'yocto': 'y',
+}
+
+
+def bidsify_units(orig_unit):
+    """
+    Read the input unit of measure and use the dictionary of aliases
+    to bidsify its value.
+    It is possible to make simple conversions
+
+    Parameters
+    ----------
+    unit: string
+        Unit of measure, might or might not be BIDS compliant.
+
+    Returns
+    -------
+    new_unit: str
+        BIDSified alias of input unit
+
+    Notes
+    -----
+    This function should implement a double check, one for unit and
+    the other for prefixes (e.g. "milli"). However, that is going to be tricky,
+    unless there is a weird way to multiply two dictionaries together.
+    """
+    # call prefix and unit dicts
+    new_unit = ""
+    # for every unit alias in the dict
+    for u_key in unit_aliases.keys():
+        unit = orig_unit.lower()
+        # check that u_key is part of unit
+        if u_key in unit:
+            new_unit = unit[-len(u_key):]
+            # check that it is at the end
+            if new_unit == u_key:
+                # leave only the prefix
+                unit = unit.replace(u_key, "")
+                # check prefix in not null
+                if unit != "":
+                    # for every prefix alias
+                    for p_key in prefix_aliases.keys():
+                        # if the alias is the same
+                        if p_key == unit:
+                            # return the units bids corrected both for prefix and unit
+                            return prefix_aliases[p_key] + new_unit
+                    # return the unit bids corrected but not the prefix
+                    LGR.warning(f'The given unit prefix {unit} does not have aliases, '
+                                f'passing it as is')
+                    return unit + new_unit
```

Comment:
Since all that's left _should_ be the prefix, you should just be able to try and get it from the dictionary (rather than iterating over the dictionary):

```python
return prefix_aliases.get(p_key, '') + new_unit
```

You can build in a check for if that `.get()` call returns `''` and raise the warning if you'd like, too.

Reply:
done